NitroxClient/GameLogic/PlayerModelBuilder/PlayerModelBuildContext.cs

```diff
@@ -0,0 +1,51 @@
+﻿using NitroxClient.GameLogic.PlayerModelBuilder.Abstract;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.PlayerModelBuilder
+{
+    // Use this context object to develop a fluent api for setting up your player model transformations.
+    // EX: Create a TankModelBuilder to handle the regular tank and then add a method here called 'WithRegularTank()'
+    // that instantiates the TankModelBuilder, passes it to the LinkBuilder method, and then returns 'this'
+    public class PlayerModelBuildContext
+    {
+        private GameObject modelGeometry;
+
+        public PlayerModelBuildContext(GameObject modelGeometry)
+        {
+            this.modelGeometry = modelGeometry;
+        }
+
+        public IPlayerModelBuilder RootBuilder { get; private set; }
+
+        public PlayerModelBuildContext WithPing()
+        {
+            PlayerPingBuilder builder = new PlayerPingBuilder();
+            LinkBuilder(builder);
+            return this;
+        }
+
+        public PlayerModelBuildContext WithRegularDiveSuit()
+        {
+            RegularDiveSuitBuilder builder = new RegularDiveSuitBuilder(modelGeometry);
+            LinkBuilder(builder);
+            return this;
+        }
+
+        public PlayerModelBuildContext WithRebreather()
+        {
+            PlayerRebreatherModelBuilder modelBuilder = new PlayerRebreatherModelBuilder(modelGeometry);
+            LinkBuilder(modelBuilder);
+            return this;
+        }
+
+        private void LinkBuilder(BasePlayerModelBuildHandler buildHandler)
```

Comment:
I assume all (future) `With` methods are supposed to be safe from any ordering? This is prepending `BuildHandler`s backwards.

Reply:
Yes - ordering should be irrelevant by implementation.